zebra-consensus/src/script.rs

```diff
@@ -12,11 +12,181 @@
+                block.transactions.into_iter()
+            })
+            .map(|transaction| {
+                transaction
+                    .inputs()
+                    .iter()
+                    .cloned()
+                    .collect::<Vec<transparent::Input>>()
+            })
+            .flat_map(|inputs| inputs.into_iter());
+
+        assert!(inputs.any(|input| matches!(input, transparent::Input::PrevOut { .. })));
+
+        let state_setup_transcript =
+            zebra_test::vectors::MAINNET_BLOCKS
+                .range(0..=10)
+                .map(|(_, block_bytes)| {
+                    let block: Arc<Block> = block_bytes.zcash_deserialize_into().unwrap();
+                    let hash = block.hash();
+
+                    let request = zebra_state::Request::CommitFinalizedBlock { block };
+                    let response = Ok::<_, TransError>(zebra_state::Response::Committed(hash));
+
+                    (request, response)
+                });
+        let state_setup_transcript = Transcript::from(state_setup_transcript);
+
+        let config = zebra_state::Config::ephemeral();
+        let network = Network::Mainnet;
+
+        let state = zebra_state::init(config, network);
+
+        /// SPANDOC: Setup the state by commiting all the necessary context
+        state_setup_transcript.check(state.clone()).await?;
+
+        let branch = NetworkUpgrade::Overwinter.branch_id().unwrap();
+        let script_service = super::Verifier::new(state, branch);
+        let script_transcript =
+            zebra_test::vectors::MAINNET_BLOCKS
+                .range(1..10)
+                .flat_map(|(_, block_bytes)| {
+                    let block: Block = block_bytes.zcash_deserialize_into().unwrap();
+
+                    block
+                        .transactions
+                        .into_iter()
+                        .flat_map(|transaction| {
+                            let inds = transaction
+                            .inputs()
+                            .iter()
+                            .enumerate()
+                            .filter(
+                                |(_, input)| !matches!(input, transparent::Input::Coinbase { .. }),
+                            )
+                            .map(|(ind, _)| ind)
+                            .collect::<Vec<_>>();
+
+                            inds.into_iter()
+                                .map(move |input_index| (input_index, transaction.clone()))
+                        })
+                        .map(|(input_index, transaction)| super::Request {
+                            input_index,
+                            transaction,
+                        })
+                        .map(|request| (request, Ok::<_, TransError>(())))
+                });
+        let script_transcript = Transcript::from(script_transcript);
+        script_transcript.check(script_service).await?;
+
+        Ok(())
+    }
+}
```

Comment:
Let's also test invalid inputs.
If there are any critical edge cases, let's test them as well.

Here's a testing checklist I use:
https://github.com/ZcashFoundation/zebra/pull/1145/files#diff-6c4c5d0b829ea5f093f27ae53dc00ebc85219486793084ccc250befec0526289R49

Reply:
I don't think it makes sense to test invalid inputs here, because this code isn't responsible for the verification itself -- that's the responsibility of the `zebra-script` crate -- it's responsible for the plumbing to asynchronously connect that verification code with the inputs it requires.